docs/dev/api.rst

```diff
@@ -0,0 +1,85 @@
+Malcolm API
```

Comment:
Don't want to check in any generated files... If you create it as ./build/api.rst and make a relative path in index.rst and then it will already be there.